examples/02-category/categories/pm.md

```diff
@@ -0,0 +1,11 @@
+---
+layout: home
+title: Porsche & Mercedes
+permalink: /porsche+mercedes
+pagination: 
+  enabled: true
+  categories:
+    - porsche
+    - mercedes
+  filter: any
```

Comment:
First of all, let's name the value `matching` that takes the values `{any, all}`. It is just easier to understand somehow than filter.

Secondly I don´t like the placement of this new `filter` option here as it sits directly on the `pagination` element and doesn't have any meaning there unless you use categories or tags. I'd much rather that we support a more complex version of the 'categories/tags' structure, one where it can be an array (and works like it currently does) or it can be a hash where we can set 'values' and 'include' on there. Think it makes it clearer what the feature does.

meaning:

```
categores:
  - porche
  - mercedes
```
or
```
categores:
   values:
     - porche
     - mercedes
   matching: all
```
```
categores:
   values:
     - porche
     - mercedes
   matching: any
```